GDAXSharp/WebSocket/WebSocket.cs

```diff
@@ -0,0 +1,157 @@
+        }
+
+        private CancellationToken CancellationToken { get; }
+
+        private List<ProductType> ProductTypes { get; set; }
+
+        private List<ChannelType> ChannelTypes { get; set; }
+
+        private WebSocket4Net.WebSocket WebSocketFeed { get; set; }
+
+        public void Start(List<ProductType> productTypes, List<ChannelType> channelTypes = null)
+        {
+            if (WebSocketFeed != null && WebSocketFeed.State != WebSocketState.Closed)
+            {
+                throw new GDAXSharpHttpException($"Websocket needs to be in the closed state, current state is {WebSocketFeed.State}");
+            }
+
+            if (productTypes.Count == 0)
+            {
+                throw new ArgumentException("You must specify at least one product type");
+            }
+
+            ProductTypes = productTypes;
+            ChannelTypes = channelTypes;
+
+            var socketUrl = SandBox ? SandBoxApiUri : ApiUri;
+
+            WebSocketFeed = new WebSocket4Net.WebSocket(socketUrl);
+            WebSocketFeed.Closed += WebSocket_Closed;
+            WebSocketFeed.Error += WebSocket_Error;
+            WebSocketFeed.MessageReceived += WebSocket_MessageReceived;
+            WebSocketFeed.Opened += WebSocket_Opened;
+            WebSocketFeed.Open();
+        }
+
+        private void WebSocket_Opened(object sender, EventArgs e)
+        {
+            var channels = new List<Channel>();
+            if (ChannelTypes == null || ChannelTypes.Count == 0)
+            {
+                foreach (ChannelType channelType in Enum.GetValues(typeof(ChannelType)))
+                {
+                    channels.Add(new Channel(channelType, ProductTypes));
+                }
+            }
+            else
+            {
+                foreach (var channelType in ChannelTypes)
+                {
+                    channels.Add(new Channel(channelType, ProductTypes));
+                }
+            }
+
+            var timeStamp = Clock.GetTime().ToTimeStamp();
+            var json = JsonConvert.SerializeObject(new TickerChannel
+            {
+                Type = ActionType.Subscribe,
+                ProductIds = ProductTypes,
+                Channels = channels,
+                Timestamp = timeStamp.ToString("F0", CultureInfo.InvariantCulture),
+                Key = Authenticator.ApiKey,
+                Passphrase = Authenticator.Passphrase,
+                Signature = ComputeSignature(HttpMethod.Get, Authenticator.UnsignedSignature, timeStamp, "/users/self/verify", null)
+            });
+
+            WebSocketFeed.Send(json);
+        }
+
+        private void WebSocket_MessageReceived(object sender, MessageReceivedEventArgs e)
+        {
+            if (CancellationToken.IsCancellationRequested)
+            {
+                WebSocketFeed.Close();
+                return;
+            }
+
+            var json = e.Message;
+            var response = JsonConvert.DeserializeObject<BaseMessage>(json);
```

Comment:
seeing an error being thrown here when you leave the list of `Channels` as null.

````
Newtonsoft.Json.JsonSerializationException occurred
  Message=Error converting value "last_match" to type 'GDAXSharp.WebSocket.Types.ResponseType'. Path 'type'
````